test/lib/ufe/testParentCmd.py

```diff
@@ -244,57 +266,36 @@ def testParentToProxyShape(self):
             sphereItem = ufe.Hierarchy.createItem(spherePath)
-    
+
+            # get the USD stage
+            stage = mayaUsd.ufe.getStage(str(shapeSegment))
+
+            # check GetLayerStack behavior
+            self.assertEqual(stage.GetLayerStack()[0], stage.GetSessionLayer())
+            self.assertEqual(stage.GetEditTarget().GetLayer(), stage.GetSessionLayer())
+
+            # set the edit target to the root layer
+            stage.SetEditTarget(stage.GetRootLayer())
+            self.assertEqual(stage.GetEditTarget().GetLayer(), stage.GetRootLayer())
+
             # The sphere is not a child of the proxy shape.
             shapeHier = ufe.Hierarchy.hierarchy(shapeItem)
             shapeChildren = shapeHier.children()
             self.assertNotIn("pSphere1", childrenNames(shapeChildren))
-    
+
             # Get its world space transform.
             sphereT3d = ufe.Transform3d.transform3d(sphereItem)
             sphereWorld = sphereT3d.inclusiveMatrix()
             sphereWorldListPre = matrixToList(sphereWorld)
-    
+
             # Parent sphere to proxy shape in absolute mode (default), using UFE
-            # path strings.
-            cmds.parent("|mayaUsdProxy1|mayaUsdProxyShape1,/pCylinder1/pCube1/pSphere1", "|mayaUsdProxy1|mayaUsdProxyShape1")
-    
-            # Confirm that the sphere is now a child of the proxy shape.
-            shapeChildren = shapeHier.children()
-            self.assertIn("pSphere1", childrenNames(shapeChildren))
-    
-            # Undo: the sphere is no longer a child of the proxy shape.
-            cmds.undo()
-    
-            shapeChildren = shapeHier.children()
-            self.assertNotIn("pSphere1", childrenNames(shapeChildren))
-    
-            # Redo: confirm that the sphere is again a child of the proxy shape.
-            cmds.redo()
-    
-            shapeChildren = shapeHier.children()
-            self.assertIn("pSphere1", childrenNames(shapeChildren))
-    
-            # Confirm that the sphere's world transform has not changed.  Must
-            # re-create the item, as its path has changed.
-            sphereChildPath = ufe.Path(
-                [shapeSegment, usdUtils.createUfePathSegment("/pSphere1")])
-            sphereChildItem = ufe.Hierarchy.createItem(sphereChildPath)
-            sphereChildT3d = ufe.Transform3d.transform3d(sphereChildItem)
-    
-            sphereWorld = sphereChildT3d.inclusiveMatrix()
-            assertVectorAlmostEqual(
-                self, sphereWorldListPre, matrixToList(sphereWorld), places=6)
-    
-            # Undo.
-            cmds.undo()
-    
-            shapeChildren = shapeHier.children()
-            self.assertNotIn("pSphere1", childrenNames(shapeChildren))
+            # path strings.Expect the exception happens
+            with self.assertRaises(RuntimeError):
+                cmds.parent("|mayaUsdProxy1|mayaUsdProxyShape1,/pCylinder1/pCube1/pSphere1", "|mayaUsdProxy1|mayaUsdProxyShape1")
```

Comment:
@ppt-adsk This test would have passed prior to applying the rules but now it fails because of third rule we have in place:

` Cannot reparent with definitions or opinions on other layers. Opinions exist in [simpleHierarchy.usda],[simpleHierarchy-session.usda]`

I am going to create a new PR to move the deleted part under `with self.assertRaises(RuntimeError):`. I really shouldn't have removed the code now that I think more about it.